test/e2e/framework/kubelet/config.go

```diff
@@ -30,11 +32,65 @@ import (
 	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
 	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
 	kubeletconfigscheme "k8s.io/kubernetes/pkg/kubelet/apis/config/scheme"
-
 	"k8s.io/kubernetes/test/e2e/framework"
 	e2ekubectl "k8s.io/kubernetes/test/e2e/framework/kubectl"
+
+	"sigs.k8s.io/yaml"
 )
 
+func getKubeletConfigFilePath() (string, error) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("failed to get current working directory: %v", err)
+	}
+
+	// DO NOT name this file "kubelet" - you will overwrite the kubelet binary and be very confused :)
+	return filepath.Join(cwd, "kubelet-config"), nil
+}
+
+// GetCurrentKubeletConfigFromFile returns the current kubelet configuration specified under /var/lib/kubelet/config.yaml.
+// This method should only run together with e2e node tests, meaning the test executor and the cluster nodes is the
```

Comment:
can we enforce this in code somehow?

Reply:
I do not see an easy way to do it, but once you will use it under the regular cluster and will see that it does not work, probably you will need to read method comments :)

Reply:
*IF* we move into the e2e_node portion this should be more obvious. Not sure if this part of the codebase is meant to be phased out (and just merge into e2e), though